### PR TITLE
Add build subcommand

### PR DESCRIPTION
**Pull request by b-deam**
With this commit we introduce a new `build` subcommand,
allowing users to simply build Elasticsearch (and plugins) for
a given target operating system and architecture.

Relates https://github.com/elastic/rally/pull/1570, https://github.com/elastic/rally-teams/pull/75/
Closes https://github.com/elastic/rally/issues/1462

### CODE REVIEW

File: esrally/mechanic/mechanic.py [hunk 1]
```diff
@@ -33,6 +33,14 @@
 METRIC_FLUSH_INTERVAL_SECONDS = 30
 
 
+def build(cfg):
```
Comment (b-deam) [inline esrally/mechanic/mechanic.py:36]: This code is _very_ similar to the `download` function, but I given their entirely seperate concerns I thought it still makes sense for these to be distinct functions.

**Comment by b-deam** [comment]
Weird, it first fails on:
```
esrally race --track="tsdb" --challenge="downsample" --track-repository="/var/lib/jenkins/workspace/elastic+rally+pull-request+rally-tracks-compat/.rally/benchmarks/tracks/rally-tracks-compat" --track-revision="master" --configuration-name="pytest" --enable-assertions --kill-running-processes --on-error="abort" --pipeline="benchmark-only" --target-hosts="127.0.0.1:19200" --test-mode
```

I can repro this locally on `master`:
```
$ esrally install --quiet --http-port=29200 --node=rally-node --master-nodes=rally-node --car=4gheap,trial-license --seed-hosts="127.0.0.1:29300" --revision=current 
{
  "installation-id": "5afc6106-5f08-407a-9304-c21614713e85"
}

$ esrally start --installation-id 5afc6106-5f08-407a-9304-c21614713e85 --race-id 2000

$ esrally race --track="tsdb" --challenge="downsample" --track-revision="master" --configuration-name="pytest" --enable-assertions --kill-running-processes --on-error="abort" --pipeline="benchmark-only" --target-hosts="127.0.0.1:19200" --test-mode
[...]
2022-09-13 03:13:23,694 ActorAddr-(T|:53103)/PID:24086 esrally.actor ERROR Worker[0] has detected a benchmark failure. Notifying master...
Traceback (most recent call last):

  File "/Users/bradleydeam/perf/github.com/b-deam/rally/esrally/driver/runner.py", line 109, in runner_for
    return __RUNNERS[operation_type]

KeyError: 'downsample'


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/Users/bradleydeam/.pyenv/versions/3.8.10/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)

  File "/Users/bradleydeam/perf/github.com/b-deam/rally/esrally/driver/driver.py", line 1707, in __call__
    loop.run_until_complete(self.run())

  File "/Users/bradleydeam/.pyenv/versions/3.8.10/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()

  File "/Users/bradleydeam/perf/github.com/b-deam/rally/esrally/driver/driver.py", line 1735, in run
    schedule = schedule_for(task_allocation, params_per_task[task])

  File "/Users/bradleydeam/perf/github.com/b-deam/rally/esrally/driver/driver.py", line 2207, in schedule_for
    runner_for_op = runner.runner_for(op.type)

  File "/Users/bradleydeam/perf/github.com/b-deam/rally/esrally/driver/runner.py", line 111, in runner_for
    raise exceptions.RallyError(f"No runner available for operation-type: [{operation_type}]")

esrally.exceptions.RallyError: No runner available for operation-type: [downsample]
```

Secondly, it fails on:
```
esrally race --track="elastic/security" --challenge="security-indexing-querying" --track-revision="master" --configuration-name="pytest" --enable-assertions --kill-running-processes --on-error="abort" --pipeline="benchmark-only" --target-hosts="127.0.0.1:29200" --test-mode --track-params="number_of_replicas:0,query_warmup_time_period:1,query_time_period:1,workflow_time_interval:1,think_time_interval:1"
```

But I can repro this failure locally.

**Comment by b-deam** [comment]
Ah, the `downsample` failure is because the [Rally Track includes the new runner](https://github.com/elastic/rally-tracks/pull/316/files), but Rally doesn't yet have [the PR merged](https://github.com/elastic/rally/pull/1574).

**Comment by b-deam** [comment]
This PR will fail ITs until https://github.com/elastic/rally-teams/pull/75 is merged.

File: docs/command_line_reference.rst [hunk 2]
```diff
@@ -83,6 +83,27 @@ This subcommand is needed for :doc:`tournament mode </tournament>` and its usage
...
+
+    esrally build --revision="@2022-07-21" --elasticsearch-plugins=analysis-icu --target-arch aarch64 --target-os linux  --quiet
+
+This will build Elasticsearch and the plugin ``analysis-icu`` for an ARM equipped Linux system, from the last commit on 2022-07-21. See :ref:`revision for more information<clr_revision>`.
+
+Because ``--quiet`` is specified, Rally will suppress all non-essential output (banners, progress messages etc.) and only return the location of the binary on the local machine after it has built it::
+
+    {
+      "elasticsearch": "/.rally/benchmarks/distributions/src/elasticsearch-ec774626230-linux-aarch64.tar.gz",
+      "analysis-icu": "file:///.rally/benchmarks/distributions/src/analysis-icu-ec774626230.zip"
```
Comment (dliappis) [inline docs/command_line_reference.rst:103]: D'oh about the `file:///` :)

**Review by dliappis — APPROVED** [review]
LGTM, thank you! This works beautifully in combination with https://github.com/elastic/rally-teams/pull/75/files.

**Comment by b-deam** [comment]
@elasticmachine run rally/it-python310

**Pull request merged**